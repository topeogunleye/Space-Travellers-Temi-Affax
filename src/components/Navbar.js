import React from 'react';
import { NavLink } from 'react-router-dom';
import planet from '../images/planet.png';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/Missions',
      text: 'Missions',
    },
    {
      id: 3,
      path: '/MyProfile',
      text: 'MyProfile',
    },
  ];

  return (
    <header className="header">
      <nav>
        <img src={planet} alt="planet logo" />
        <h1>Space Travellers Hub</h1>
        <ul>
          {links.map((item) => (
            <li key={item.id}>
              <NavLink to={item.path} className="navlink">
                {item.text}
              </NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
};

export default Navbar;
