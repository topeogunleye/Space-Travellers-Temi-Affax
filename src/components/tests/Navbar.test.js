import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Navbar from '../Navbar';

describe('the nabvar display ', () => {
  test('rendering of the Navbar', () => {
    const tree = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
