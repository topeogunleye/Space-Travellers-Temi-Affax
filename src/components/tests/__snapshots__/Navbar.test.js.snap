// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the nabvar display  rendering of the Navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class=""
      >
        <nav
          class="md:flex flex-col md:flex-row justify-between items-center"
        >
          <div
            class="flex  justify-between items-center w-72"
          >
            <img
              alt="planet logo"
              class="w-12"
              src="planet.png"
            />
            <h1
              class="font-bold text-xl"
            >
              Space Travellers Hub
            </h1>
          </div>
          <ul
            class="flex justify-between items-center w-56 text-blue-600 divide-x-2-black "
          >
            <li
              class=""
            >
              <a
                aria-current="page"
                class="navlink active"
                href="/"
                style="text-decoration: underline;"
              >
                Rockets
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="navlink"
                href="/Missions"
                style="text-decoration: #545e6f;"
              >
                Missions
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="navlink"
                href="/MyProfile"
                style="text-decoration: #545e6f;"
              >
                MyProfile
              </a>
            </li>
          </ul>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class=""
    >
      <nav
        class="md:flex flex-col md:flex-row justify-between items-center"
      >
        <div
          class="flex  justify-between items-center w-72"
        >
          <img
            alt="planet logo"
            class="w-12"
            src="planet.png"
          />
          <h1
            class="font-bold text-xl"
          >
            Space Travellers Hub
          </h1>
        </div>
        <ul
          class="flex justify-between items-center w-56 text-blue-600 divide-x-2-black "
        >
          <li
            class=""
          >
            <a
              aria-current="page"
              class="navlink active"
              href="/"
              style="text-decoration: underline;"
            >
              Rockets
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="navlink"
              href="/Missions"
              style="text-decoration: #545e6f;"
            >
              Missions
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="navlink"
              href="/MyProfile"
              style="text-decoration: #545e6f;"
            >
              MyProfile
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
